services/director/src/simcore_service_director/producer.py

```diff
@@ -1033,42 +1066,40 @@ async def stop_service(app: web.Application, node_uuid: str, save_state: bool) -
             if not "3d-viewer" in service_details["service_host"]
             else "",
         )
-        log.debug("saving state of service %s...", service_host_name)
+
+        # If state save is enforced, it will fail if not completed
         if save_state:
+            log.debug("saving state of service %s...", service_host_name)
             try:
-                session = app[APP_CLIENT_SESSION_KEY]
-                service_url = "http://" + service_host_name + "/" + "state"
-                async with session.post(
-                    service_url,
-                    timeout=ServicesCommonSettings().director_dynamic_service_save_timeout,
-                ) as response:
-                    if 199 < response.status < 300:
-                        log.debug(
-                            "service %s successfully saved its state", service_host_name
-                        )
-                    else:
-                        log.warning(
-                            "service %s does not allow saving state, answered %s",
-                            service_host_name,
-                            await response.text(),
-                        )
-            except ClientConnectionError:
-                log.warning(
-                    "service %s could not be contacted, state not saved",
-                    service_host_name,
+                await _save_service_state(
+                    service_host_name, session=app[APP_CLIENT_SESSION_KEY]
                 )
+            except ClientResponseError as err:
+                raise ServiceStateSaveError(
+                    node_uuid,
+                    reason=f"service {service_host_name} rejected to save state, "
+                    f"responded {err.message} (status {err.status})",
+                ) from err
```

Comment:
Just curious: Up until now, we logged warnings. What happens when the error is raised, does the director-v0 resume execution? How logs

Reply:
exceptions like these raised in a handler are converted into 500s (or sometimes mapped to more concrete error codes) and also logged by the middleware on the service before responding.

But now that you mention it, I will double check because this is a very old service and that feature might be missing ...
thanks for raising that up!

Reply:
double checkd. Yes, the exception is transformed into a 500 and logged